kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go

```diff
@@ -222,35 +206,50 @@ func (r *Reconciler) createSubscribableStatus(subscribable *eventingduckv1alpha1
 }
 
 // newConfigFromKafkaChannels creates a new Config from the list of kafka channels.
-func (r *Reconciler) newChannelConfigFromKafkaChannel(c *v1alpha1.KafkaChannel) *multichannelfanout.ChannelConfig {
-	channelConfig := multichannelfanout.ChannelConfig{
+func (r *Reconciler) newChannelConfigFromKafkaChannel(c *v1alpha1.KafkaChannel) *dispatcher.ChannelConfig {
+	channelConfig := dispatcher.ChannelConfig{
 		Namespace: c.Namespace,
 		Name:      c.Name,
 		HostName:  c.Status.Address.Hostname,
 	}
 	if c.Spec.Subscribable != nil {
-		newSubs := make([]eventingduckv1beta1.SubscriberSpec, len(c.Spec.Subscribable.Subscribers))
-		for i, source := range c.Spec.Subscribable.Subscribers {
-			source.ConvertTo(context.TODO(), &newSubs[i])
-			fmt.Printf("Converted \n%+v\n To\n%+v\n", source, newSubs[i])
-			fmt.Printf("Delivery converted \n%+v\nto\n%+v\n", source.Delivery, newSubs[i].Delivery)
-		}
-		channelConfig.FanoutConfig = fanout.Config{
-			AsyncHandler:  true,
-			Subscriptions: newSubs,
+		newSubs := make([]dispatcher.Subscription, 0, len(c.Spec.Subscribable.Subscribers))
+		for _, source := range c.Spec.Subscribable.Subscribers {
+
+			// Extract retry configuration
+			var retryConfig *kncloudevents.RetryConfig
```

Comment:
And all the code here can be just saved using that fanout config i sad in previous comment